### DISCUSSION TITLE
Hierarchical Menu - Modify how to detect an item is a "parent" or not ?

### DISCUSSION THREAD

**Discussion by Pochwar** [algolia/instantsearch#5456] [Q&A] [answered]
Hello, I'm using the hierachcal-menu widget in my app, and it's a great tool, but I have an edge case where "parent detection" fails.

For reasons related to the business, some categories are not displayed in front but only in back-office. But the depth of the tree is calculated according to all the categories, even those that are not displayed on the front, and I use this depth to generate dynamically the attributes ('categories.lvl0',  'categories.lvl1' ...) that I provide to the widget.

So sometimes (I have different organizations with different category trees) the tree in front is shallower than the calculated depth based on all the categories displayed in back office (and it's complicated to change that) and the result is that some items in the hierarchical menu are considered as parent when they are not because `item.data` is not `null` but is an empty array.

So my question is : Do you think it would be possible to modify the way an item is considered as a parent?

As I use `vue-instantsearch` here are the modifications that I suggest:

File: HierarchicalMenuList.vue
- current:
```
<li
  v-for="item in items"
  :key="item.value"
  :class="[
    suit('item'),
    item.data && suit('item', 'parent'),
    item.isRefined && suit('item', 'selected'),
  ]"
>
```
- suggestion: replace `item.data` with a method `hasChildren(item.data)`
```
<li
  v-for="item in items"
  :key="item.value"
  :class="[
    suit('item'),
    hasChildren(item.data) && suit('item', 'parent'),
    item.isRefined && suit('item', 'selected'),
  ]"
>
```
and the `hasChildren()` method:
```
hasChildren(data) {
  return !(!data || data.length === 0);
},
```

**Comment by Haroenv** [comment, marked as answer]
Yes, this would make sense and is already the case in the other flavours. I've opened a PR for that here: https://github.com/algolia/instantsearch/pull/5458

**Reply by Pochwar** [reply]
Nice, thx for your quick answer

**Reply by Haroenv** [reply]
this will be released in the next version, for Vue InstantSearch that is 4.8.1
